packages/core/src/api/nodeUtil.test.ts

```diff
@@ -0,0 +1,361 @@
+import { describe, expect, it, vi, beforeEach, afterEach } from "vitest";
+
+import { setupTestEnv } from "./blockManipulation/setupTestEnv.js";
+import { getBlocksChangedByTransaction } from "./nodeUtil.js";
+import { Transaction } from "prosemirror-state";
+import { BlockNoteEditor } from "../editor/BlockNoteEditor.js";
+import { Step } from "prosemirror-transform";
+
+const getEditor = setupTestEnv();
+
+describe("Test getBlocksChangedByTransaction", () => {
+  let transaction: Transaction;
+  let editor: BlockNoteEditor;
+  let originalDispatch: typeof editor.dispatch;
+
+  beforeEach(() => {
+    transaction = undefined as unknown as Transaction;
```

Comment:
not easier to test the onchange event directly instead of patching like this?

Reply:
Likely easier, but the test is not for the change event, but to see if a transaction can give us the changes. I'd want it isolated